xesmf/cf.py

```diff
@@ -0,0 +1,158 @@
+"""
+Decode longitude and latitude of an `xarray.Dataset` or a `xarray.DataArray`
+according to CF conventions.
+"""
+
+import warnings
+
+CF_SPECS = {
+    'lon': {
+        'name': ['lon', 'longitude'],
+        'units': ['degrees_east', 'degree_east', 'degree_E', 'degrees_E',
```

Comment:
Checking the units and other attributes seems a bit too heavy for xesmf? A "CF-checker" is better for a standalone package or pushed to upstream like [xarray.decode_cf](http://xarray.pydata.org/en/stable/generated/xarray.decode_cf.html)

Reply:
I agree with you that a CF checker for xarray is globally missing. 
However, xarray is intended to remain of general purpose, and not to introduce notions of geographic and physical spatialisation.

As for the fact that these attributes are checked is the strict application of the CF conventions for [longitude ](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.7/cf-conventions.html#longitude-coordinate) and latitude. 
This prevent focusing on names.